Library/brew.sh

```diff
@@ -160,6 +162,39 @@ EOS
   esac
 fi
 
+if [[ -n "$HOMEBREW_ANALYTICS" ]]
+then
+  # User UUID file. Used for Homebrew user counting. Can be deleted and
+  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -r "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(cat "$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+  else
+    HOMEBREW_ANALYTICS_USER_UUID="$(uuidgen)"
+    echo "$HOMEBREW_ANALYTICS_USER_UUID" > "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
```

Comment:
> Otherwise, uuid itself is an identifier which can be used to track and pin point user.

I think that’s the point: if we want to e.g. count how many users we have we need to have one ID per user. If we want to know how many users installed formula F with options `--foo` and `--bar` but one of them installs it 10 times a day we need an user id to detect that.